### PR TITLE
Adding in the ability to use a custom logger in place of the default one

### PR DESCRIPTION
**Pull request by StabbyCutyou**
This is to address #39 

I figured since you already had the config abstraction, the best thing to do would be to layer this into the existing logoutput options, but to leave those in to maintain compatibility.

### CODE REVIEW

**Comment by ErikDubbelboer** [comment]
If this is added I vote to remove `config.LogOutput`. Having both is a little bit confusing and `LogOutput` is not really needed if you have `Logger`. It would break API compatibility though.

**Comment by StabbyCutyou** [comment]
Thats why I left it in - since theres no fantastic way to set a dependency version in a standard way, I didn't want to break the API compatibility. I figured it was safer to just do it this way.

File: config.go [hunk 1]
```diff
@@ -136,6 +137,10 @@ type Config struct {
 	// LogOutput is the writer where logs should be sent. If this is not
 	// set, logging will go to stderr by default.
 	LogOutput io.Writer
+	// Logger is a custom logger which you provide. If Logger is set, it will use this for
+	// the internal logger. If this is not set, it will fall back to the behavior for
+	// LogOutput is not set, logging will go to stderr by default.
+	Logger *log.Logger
```
Comment (ryanuber) [inline config.go]: Can we add a line break between this and the field above? The description could also be cleaned up to be more clear about the LogOutput writer being used when Logger is nil. We should also add a similar note to the LogOutput fie.

**Comment by ryanuber** [comment]
This seems mostly fine to me. It would make sense to return an error if both LogOutput and Logger are non-nil rather than to silently pick one. If we can also add a test case for this, that would be great!

**Comment by StabbyCutyou** [comment]
Happy to add test cases - Didn't wanna put that time in yet in case the PR
was rejected, but can definitely add those in.

If people have suggestions for better comments, I'm happy to incorporate
that as well.

**Comment by StabbyCutyou** [comment]
Changed the comments around a bit, but can edit them some more if you like.

Added the test case for having both set return an error.

**Comment by slackpad** [comment]
This LGTM - thanks!

**Pull request merged**